File: search_matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        m = len(matrix[0])
        n = len(matrix)
        i = 0
        j = (m*n) - 1
        while(i <= j):
            mid = (i + j) // 2
            if(mid != 0):
                x = matrix[mid//m][mid%m]
            else:
                x = matrix[0][0]    #Avoiding division by zero
            if(x == target):
                return True
            elif(x < target):
                i = mid + 1
            else:
                j = mid - 1
        return False

File: search_matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix[0].size(), n = matrix.size();
        int i = 0, j = (m*n)-1;
        while(i <= j)
        {
            int mid = (i + j) / 2, x;
            if(mid != 0)
                x = matrix[mid/m][mid%m];
            else
                x = matrix[0][0];    //Avoiding division by zero
            if(x == target)
                return true;
            else if(x < target)
                i = mid + 1;
            else
                j = mid - 1;
        }
        return false;
    }
};